### DISCUSSION TITLE
Globbing and edges

### DISCUSSION THREAD

**Discussion by canelhasmateus** [terrastruct/d2#1555] [Q&A] [answered]
I would like to understand how to target every edge coming in / out from filtered objects.

In this [example](https://play.d2lang.com/?script=fJHdCsIwDIXv8xSHFaQW9AF6Ifj3FGMMweCFuI3OO-m7S9LSKv7cjJPlnOYL2WK1wTyNV6ZdlfqVcl_locojJe3xIECP4dHO441VdxSJ8omSpbRSCegGa7mFR6PQ0ynwcG9eumnjz36U522BccvWdchz3oOBzw1Fsq5s9tt7CcyDuMk4MRhgkZELPRn5bfsyNhu_DzaCagDNKEL_L5MBUirSMwAA__8%3D&), for some subset of the objects, I want to set the color of every incoming edge as green, and every outgoing edge as blue.

It works as long as targeting specific objects, as seen in the uncommented part.
 
However, If there were multiple `spokes`, specifying each pair would be a pain, and it would be easier to specify the object subset by filtering.
I could not find such example in the documentation, and figured that the above would be the syntax, but it does not work. 
Uncommenting it fails with `indexed edge does not exist`.

How would one glob edges from globbed objects?
Note: although the exemplifying usecase is legit, my own use case is trying to work around the [lack of links in edges](https://github.com/terrastruct/d2/issues/1497)

```d2
A -> spoke
B -> spoke
spoke -> C
spoke -> D
spoke -> E

spoke: {
  class: [someclass]
}

classes: {
  someclass: {
    style.fill: "transparent"
    style.stroke: "transparent"
  }
}

(spoke -> *)[*]: {
  style.stroke: "red"
}
(* -> spoke)[*]: {
  style.stroke: "green"
}

# *: {
#  &class: someclass
#
#  (_ -> *)[*]: {
#    style.stroke: "red"
#  }
#  
#  (* -> _)[*]: {
#    style.stroke: "green"
#  }
# }
```

**Comment by alixander** [comment, marked as answer]
@canelhasmateus 

this would be a case of more filters, which we'll add soon

```d2
(* -> *)[*]: {
  # filter for all edges that point to an object with label "spoke"
  dst: {
    &label: spoke
  }
  style.stroke: "green"
}
```
